implementations/rust/ockam/ockam_transport_tcp/src/portal/portal_receiver.rs

```diff
@@ -39,12 +42,19 @@ impl Processor for TcpPortalRecvProcessor {
         };
 
         if len != 0 {
-            let mut vec = vec![0u8; len];
-            vec.copy_from_slice(&buf[..len]);
-            let msg = PortalInternalMessage::Payload(vec);
+            loop {
```

Comment:
This loop is kind of odd. `read_buf` can never increase the buf past its capacity, so `buf.len()` will be less than `MAX_PAYLOAD_SIZE` already. So this is just `buf.split_to(buf.len())`.

So we'll always bail out after 1 iteration -- no need to loop.